RecoJets/JetPlusTracks/python/PATJetPlusTrackCorrections_cff.py

```diff
@@ -0,0 +1,93 @@
+                                         tracks = "produceTracks",
+                                         extrapolations = "trackExtrapolatorPAT"
+)
+ak4JetExtenderJPTPAT = ak4JetExtender.clone(
+                       jets = "slimmedCaloJets",
+                       jet2TracksAtCALO = "ak4JetTracksAssociatorAtCaloFaceJPTPAT",
+                       jet2TracksAtVX = "ak4JetTracksAssociatorAtVertexJPTPAT"
+)
+
+# ---------- Supported Modules
+
+from CommonTools.RecoAlgos.trackFromPackedCandidateProducer_cfi import *
+produceTracks = trackFromPackedCandidateProducer.clone()
+
+
+from RecoJets.JetPlusTracks.jetPlusTrackAddonSeedProducer_cfi import *
+JetPlusTrackAddonSeedRecoPAT = jetPlusTrackAddonSeedProducer.clone(
+    srcCaloJets = cms.InputTag("slimmedCaloJets"),
+    srcPVs = cms.InputTag('offlineSlimmedPrimaryVertices'),
+    UsePAT = True
+)
+
+
+from CommonTools.RecoAlgos.TrackWithVertexRefSelector_cfi import *
+trackWithVertexRefSelector.vertexTag = cms.InputTag('offlineSlimmedPrimaryVertices')
+trackWithVertexRefSelector.src = cms.InputTag('produceTracks')
+from RecoJets.JetProducers.TracksForJets_cff import *
+from RecoJets.Configuration.RecoTrackJets_cff import *
+ak4TrackJets.srcPVs = cms.InputTag('offlineSlimmedPrimaryVertices')
+ak4TrackJets.UseOnlyOnePV = cms.bool(True)
+
+# ---------- Module definition
+
+from RecoJets.JetPlusTracks.JetPlusTrackCorrections_cfi import *
+
+JetPlusTrackZSPCorJetAntiKt4PAT = cms.EDProducer(
+    "JetPlusTrackProducer",
+    cms.PSet(JPTZSPCorrectorAntiKt4),
+    src = cms.InputTag("slimmedCaloJets"),
+    srcTrackJets = cms.InputTag("ak4TrackJets"), 
+    srcAddCaloJets = cms.InputTag('JetPlusTrackAddonSeedRecoPAT'),
+    extrapolations = cms.InputTag("trackExtrapolatorPAT"),
+    tagName = cms.vstring('ZSP_CMSSW390_Akt_05_PU0'),
+    tagNameOffset = cms.vstring(),
+    PU = cms.int32(-1),
+    FixedPU = cms.int32(0),
+    UseZSP = cms.bool(False),
+    srcPVs = cms.InputTag('offlineSlimmedPrimaryVertices'),    
+    alias = cms.untracked.string('JetPlusTrackZSPCorJetAntiKt4'),
+    ptCUT = cms.double(15.),
+    dRcone = cms.double(0.4)
+    )
+
+JetPlusTrackZSPCorJetAntiKt4PAT.JetTracksAssociationAtVertex = "ak4JetTracksAssociatorAtVertexJPTPAT"
+JetPlusTrackZSPCorJetAntiKt4PAT.JetTracksAssociationAtCaloFace = "ak4JetTracksAssociatorAtCaloFaceJPTPAT"
+JetPlusTrackZSPCorJetAntiKt4PAT.Muons = "slimmedMuons"
+JetPlusTrackZSPCorJetAntiKt4PAT.Electrons = "slimmedElectrons"
+JetPlusTrackZSPCorJetAntiKt4PAT.JetSplitMerge = 2
+JetPlusTrackZSPCorJetAntiKt4PAT.UsePAT = True
+### ---------- Sequences
+
+# Anti-Kt
+
+PATJetPlusTrackCorrectionsAntiKt4 = cms.Sequence(
+    produceTracks*
+    trackWithVertexRefSelector*
+    trackRefsForJets*
+    ak4TrackJets*
+    JetPlusTrackAddonSeedRecoPAT*
+    trackExtrapolatorPAT*
+    ak4JetTracksAssociatorAtVertexJPTPAT*
+    ak4JetTracksAssociatorAtCaloFaceJPTPAT*
+    ak4JetExtenderJPTPAT*
+    JetPlusTrackZSPCorJetAntiKt4PAT
+    )
```

Comment:
```suggestion
PATJetPlusTrackCorrectionsAntiKt4Task = cms.Task(
    produceTracks,
    trackWithVertexRefSelector,
    trackRefsForJets,
    ak4TrackJets,
    JetPlusTrackAddonSeedRecoPAT,
    trackExtrapolatorPAT,
    ak4JetTracksAssociatorAtVertexJPTPAT,
    ak4JetTracksAssociatorAtCaloFaceJPTPAT,
    ak4JetExtenderJPTPAT,
    JetPlusTrackZSPCorJetAntiKt4PAT
    )
PATJetPlusTrackCorrectionsAntiKt4 = cms.Sequence(PATJetPlusTrackCorrectionsAntiKt4Task)
```

just in case the earlier suggestion, which included some plain English, was not clear

Reply:
I did exactly this. I put printout in JetPlusTrackProducer.cc: just number of jets. With Task I did not get this printout that means that I did not come to the JetPlusTrackProducer.cc during execution. With Sequence I get this printout.
While at the same time, Task works normally for Reco part.

Reply:
This is not the first time when I have troubles with Task which silently does nothing. And I do not have any keys to understand what is going on.

Reply:
> I did exactly this. I put printout in JetPlusTrackProducer.cc: just number of jets. With Task I did not get this printout that means that I did not come to the JetPlusTrackProducer.cc during execution. With Sequence I get this printout.

Likely this means that there is nothing in job that would consume the output of `JetPlusTrackProducer` (in such a case framework will not run a module for events). Should some other module or output module consume its output?

Reply:
@makortel @Dr15Jones 
if we have a case of an "analyzer" (some ED module that reads data, e.g. to make a printout as in this case, but is not expected to write a persisted product), is cms.Task something that can still work, or do we have to put the module in the cms.Sequence where it has to run?

Reply:
It has to go in a sequence. Things in Tasks are run on demand and EDAnalyzers do not produce anything.

Reply:
> It has to go in a sequence. Things in Tasks are run on demand and EDAnalyzers do not produce anything.

OK, thanks for clarifying.

In this specific case, it's clear that JetPlusTrackProducer is a producer.
@kodolova why is it not supposed to produce anything?
Are you referring to a test configuration where you do not have anything yet that would read the jets when running on a miniAOD input?
It sounds like this circles back to https://github.com/cms-sw/cmssw/pull/30430#pullrequestreview-452554478
- I'm missing an example test config to run the new code with PAT enabled (RecoJets/JetPlusTracks/python/PATJetPlusTrackCorrections_cff.py only gives a partial idea)